### DISCUSSION TITLE
Hello World scenario fixed

### DISCUSSION THREAD

**Discussion by oliverbarnes** [decidim/docker#67] [Show and tell]
As [it was discussed a while back](https://github.com/decidim/decidim/issues/5405), the decidim/decidim:latest image up on Docker Hub isn't working with the proposed "Hello World" docker-compose flow, because the image is an app generator, and not an app instance. The details are discussed thoroughly on that thread, so I'll skip them.

With the recent changes that were merged (#50), the generator image is now called `decidim-generator`, to avoid this confusion, and the `decidim` image is now based on Dockerfile-deploy. Dockerfile-deploy itself was updated to fix the following error:

```bash
➜ docker run -it --rm \
  -e DATABASE_USERNAME=postgres \
  -e DATABASE_PASSWORD=postgres \
  -e DATABASE_HOST=host.docker.internal \
  -e RAILS_ENV=development \
  -p 3000:3000 \
  decidim/decidim:latest-deploy
Could not locate Gemfile or .bundle/ directory
```

The update and fix was to generate the Decidim app within the build process, and remove the `ONBUILD` commands. I'm now able to get the app up and running smoothly locally. Here are the steps with an existing local postgres database, without using docker-compose so it's transparent what's going on. I'm not posting the output, but db setup, migrations and seeding all run smoothly. 

```bash
docker run -it --rm \
  -e DATABASE_USERNAME=postgres \
  -e DATABASE_PASSWORD=postgres \
  -e DATABASE_HOST=host.docker.internal \
  -e RAILS_ENV=development \
  -p 3000:3000 \
  ghcr.io/decidim/decidim:latest bundle exec rake db:setup

docker run -it --rm \
  -e DATABASE_USERNAME=postgres \
  -e DATABASE_PASSWORD=postgres \
  -e DATABASE_HOST=host.docker.internal \
  -e RAILS_ENV=development \
  -p 3000:3000 \
  ghcr.io/decidim/decidim:latest bundle exec rake db:migrate

docker run -it --rm \
  -e DATABASE_USERNAME=postgres \
  -e DATABASE_PASSWORD=postgres \
  -e DATABASE_HOST=host.docker.internal \
  -e RAILS_ENV=development \
  -p 3000:3000 \
  ghcr.io/decidim/decidim:latest bundle exec rake db:seed

docker run -it --rm \
  -e DATABASE_USERNAME=postgres \
  -e DATABASE_PASSWORD=postgres \
  -e DATABASE_HOST=host.docker.internal \
  -e RAILS_ENV=development \
  -p 3000:3000 \
  ghcr.io/decidim/decidim:latest
```

Notes: 
- `host.docker.internal` is just how Docker for mac accesses localhost on the host machine, this wouldn't be needed on linux.

**To sum it up**

The Hello World scenario is fixed, but with a different dev experience. We'll need to update documentation, and decide we want to use docker-compose to also offer to spin up the db for the dev user.

**Comment by oliverbarnes** [comment]
Update: the new `decidim-generator` and `decidim` images are now published on Github Container Registry: https://github.com/orgs/decidim/packages/container/package/decidim

They will be mirrored on Docker Hub but there's some configuration pending in order to do so.

**Comment by oliverbarnes** [comment]
https://github.com/decidim/docker/pull/78 now adds a docker-compose "Hello World" as well:

```bash
git clone git@github.com:decidim/docker.git decidim-docker
cd decidim-docker
docker-compose up
```
